aws/resource_aws_macie2_account.go

```diff
@@ -0,0 +1,159 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/macie2"
+	"github.com/hashicorp/aws-sdk-go-base/tfawserr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceAwsMacie2Account() *schema.Resource {
+	return &schema.Resource{
+		CreateWithoutTimeout: resourceMacie2AccountCreate,
+		ReadWithoutTimeout:   resourceMacie2AccountRead,
+		UpdateWithoutTimeout: resourceMacie2AccountUpdate,
+		DeleteWithoutTimeout: resourceMacie2AccountDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"finding_publishing_frequency": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.StringInSlice(macie2.FindingPublishingFrequency_Values(), false),
+			},
+			"status": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.StringInSlice(macie2.MacieStatus_Values(), false),
+			},
+			"service_role": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"updated_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceMacie2AccountCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*AWSClient).macie2conn
+
+	input := &macie2.EnableMacieInput{
+		ClientToken: aws.String(resource.UniqueId()),
+	}
+
+	if v, ok := d.GetOk("finding_publishing_frequency"); ok {
+		input.FindingPublishingFrequency = aws.String(v.(string))
+	}
+	if v, ok := d.GetOk("status"); ok {
+		input.Status = aws.String(v.(string))
+	}
+
+	var err error
+	err = resource.RetryContext(ctx, 4*time.Minute, func() *resource.RetryError {
+		_, err = conn.EnableMacieWithContext(ctx, input)
+		if err != nil {
+			if tfawserr.ErrCodeEquals(err, macie2.ErrorCodeClientError) {
+				return resource.RetryableError(err)
+			}
+
+			return resource.NonRetryableError(err)
+		}
+
+		return nil
+	})
+
+	if isResourceTimeoutError(err) {
+		_, err = conn.EnableMacieWithContext(ctx, input)
+	}
+
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error enabling Macie Account: %w", err))
+	}
+
+	d.SetId(meta.(*AWSClient).accountid)
+
+	return resourceMacie2AccountRead(ctx, d, meta)
+}
+
+func resourceMacie2AccountRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*AWSClient).macie2conn
+
+	input := &macie2.GetMacieSessionInput{}
+
+	resp, err := conn.GetMacieSessionWithContext(ctx, input)
+
+	if tfawserr.ErrCodeEquals(err, macie2.ErrCodeAccessDeniedException) {
```

Comment:
I missed this one. It should have the same error test as in the Delete operation, since the AWS API potentially returns both errors